### PR TITLE
feat: Make next_block_blob_fee functions accept an update fraction parameter 

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
closes https://github.com/alloy-rs/alloy/issues/1766

### CODE REVIEW

File: crates/consensus/src/block/header.rs [hunk 1]
```diff
@@ -214,10 +216,41 @@ impl Header {
...
+    ///
+    /// # Returns
+    ///
+    /// - `Some(u128)` containing the calculated blob fee if `next_block_excess_blob_gas` is `Some`.
+    /// - `None` if `next_block_excess_blob_gas` is `None`.
+    ///
+    /// # Panics
+    ///
+    /// - Panics if `update_fraction` is provided and not within the range [0.0, 1.0].
+    pub fn next_block_blob_fee_with_fraction(&self, update_fraction: Option<f64>) -> Option<u128> {
```
Comment (klkvr) [inline crates/consensus/src/block/header.rs:241]: I'd prefer this to not be optional

Comment (klkvr) [reply crates/consensus/src/block/header.rs:241]: and we also need this on `BlochHeader` trait

File: crates/consensus/src/block/header.rs [hunk 2]
```diff
@@ -214,10 +216,41 @@ impl Header {
...
+        let fraction = update_fraction.unwrap_or(BLOB_BASE_FEE_UPDATE_FRACTION_ELECTRA);
+
+        if !(0.0..=1.0).contains(&fraction) {
+            panic!("update_fraction must be between 0.0 and 1.0");
+        }
+
+        let excess_blob_gas = self.next_block_excess_blob_gas()?;
+        let adjusted_blob_gas = (excess_blob_gas as f64 * fraction) as u64;
+
+        Some(eip4844::calc_blob_gasprice(adjusted_blob_gas))
```
Comment (klkvr) [inline crates/consensus/src/block/header.rs:251]: we should introduce an additional helper like `calc_blob_gasprice_with_update_fraction` and use it here

**Review by klkvr — CHANGES_REQUESTED** [review]

File: crates/consensus/src/block/header.rs [hunk 3]
```diff
@@ -214,10 +214,39 @@ impl Header {
     /// Returns `None` if `excess_blob_gas` is None.
     ///
     /// See also [Self::next_block_excess_blob_gas]
+    #[deprecated(
+        since = "0.8.1",
```
Comment (DoTheBestToGetTheBest) [inline crates/consensus/src/block/header.rs:218]: not sure if  you are okay with this version the current one is 0.8.0 so for the next one is 0.8.1;

lmk if you want me to change or remove this

File: crates/consensus/src/block/header.rs (hunk 2 above)
Comment (DoTheBestToGetTheBest) [reply crates/consensus/src/block/header.rs]: > we should introduce an additional helper like `calc_blob_gasprice_with_update_fraction` and use it here

thanks for your review!

File: crates/consensus/src/block/header.rs (hunk 1 above)
Comment (DoTheBestToGetTheBest) [reply crates/consensus/src/block/header.rs]: > and we also need this on `BlochHeader` trait

thanks, everything done!

**Comment by DoTheBestToGetTheBest** [comment]
ping @klkvr

**Comment by mattsse** [comment]
@klkvr with recent 77840 changes https://github.com/alloy-rs/alloy/pull/1828 this is no longer required?

**Comment by klkvr** [comment]
yeah sorry @DoTheBestToGetTheBest, this is superseded by https://github.com/alloy-rs/alloy/pull/1828

**Pull request closed without merging**